backend/migration_tools/management/commands/migrate_to_v2.py

```diff
@@ -0,0 +1,367 @@
+        return psycopg2.connect(**db_config)
+
+    def create_tracking_table_if_not_exists(self):
+        conn = self.connect_db(self.dest_db_config)
+        cur: cursor = conn.cursor()
+        try:
+            cur.execute(
+                f"""
+                CREATE TABLE IF NOT EXISTS {self.migration_tracking_table} (
+                    id SERIAL PRIMARY KEY,
+                    migration_name VARCHAR(255) UNIQUE NOT NULL,
+                    applied_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP
+                );
+            """
+            )
+            conn.commit()
+            logger.info("Migration tracking table ensured.")
+        except psycopg2.Error as e:
+            conn.rollback()
+            logger.error(f"Error creating migration tracking table: {e}")
+        finally:
+            cur.close()
+            conn.close()
+
+    def migration_applied(self, migration_name):
+        conn = self.connect_db(self.dest_db_config)
+        cur: cursor = conn.cursor()
+        cur.execute(
+            f"""
+            SELECT COUNT(*) FROM {self.migration_tracking_table}
+            WHERE migration_name = %s;"
+            """,
+            (migration_name,),
+        )
+        result = cur.fetchone()
+        cur.close()
+        conn.close()
+        return result[0] > 0
+
+    def record_migration(self, migration_name):
+        conn = self.connect_db(self.dest_db_config)
+        cur: cursor = conn.cursor()
+        cur.execute(
+            f"""
+            INSERT INTO {self.migration_tracking_table} (migration_name)
+            VALUES (%s);
+            """,
+            (migration_name,),
+        )
+        conn.commit()
+        cur.close()
+        conn.close()
+
+    def fetch_schema_names(
+        self, schemas_to_migrate: Optional[list[str]] = None
+    ) -> list[tuple[int, str]]:
+        conn = self.connect_db(self.src_db_config)
+        cur = conn.cursor()
+
+        # Process schemas_to_migrate: trim spaces and remove empty entries
+        if schemas_to_migrate:
+            schemas_to_migrate = [
+                schema.strip() for schema in schemas_to_migrate if schema.strip()
+            ]
+
+        if schemas_to_migrate:
+            placeholders = ",".join(["%s"] * len(schemas_to_migrate))
+            query = (
+                "SELECT id, schema_name FROM account_organization WHERE "
+                f"schema_name IN ({placeholders});"
+            )
+            cur.execute(query, schemas_to_migrate)
+        else:
+            query = "SELECT id, schema_name FROM account_organization;"
+            cur.execute(query)
+
+        schema_names = [(row[0], row[1]) for row in cur.fetchall()]
+        cur.close()
+        conn.close()
+        return schema_names
+
+    def migrate_single_row(
+        self,
+        row: tuple[Any, ...],
+        dest_cursor: cursor,
+        column_names: list[str],
+        new_key_transaction: dict[str, dict[str, Any]],
```

Comment:
NIT: @muhammad-ali-e can you rename this variable or add a docstring on what this is?